torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -23,12 +23,21 @@ void addSendRpcBackward(
     const AutogradMetadata& autogradMetadata,
     std::vector<torch::Tensor>& tensors,
     const rpc::worker_id_t dst) {
+  // Attach autograd information only for tensors requiring grad.
+  std::vector<torch::Tensor> tensors_with_grad;
+  std::copy_if(
```

Comment:
Question, does`copy_if` create a deepcopy of the tensor?  It might not be idea to make a copy of  the tensors. 
I did not find a `move_if` in std. if we do not need to original vector, we might want to use `remove_if` from the original Tensor.

Reply:
@zzzwen Tensor internally has a reference counted pointer, only that pointer is copied. This is not a deep copy of the tensor.